qutebrowser/browser/webkit/webpage.py

```diff
@@ -86,6 +86,12 @@ def __init__(self, win_id, tab_id, tabdata, private, parent=None):
             self.on_save_frame_state_requested)
         self.restoreFrameStateRequested.connect(
             self.on_restore_frame_state_requested)
+        self.mainFrame().javaScriptWindowObjectCleared.connect(
+            self.inject_userjs_start)
```

Comment:
You could do `functools.partial(self.inject_userjs, load='start')` instead. (same for the others)